### ISSUE TITLE
Configure multi-instance versioning

### ISSUE THREAD

**Issue by pepopowitz** [camunda/camunda-platform-docs#1116]
Epic: https://github.com/camunda/product-hub/issues/231
Prerequisite: #1115 

Optimize versions are different than the versions of the other Camunda components. [Docusaurus recommends using multiple instances of the docs plugin](https://docusaurus.io/docs/next/docs-multi-instance#mobile-sdks-documentation) to enable users to see different versions for different docs.

### Preparation (ahead of time)
- #1167 (done in #1170)
- #1193 

### Move docs over
1. ✅ Create a "next" Optimize version
     1. ✅ #1166
          This moves a single doc into the new instance, including all configuration necessary to iteratively move more docs over afterward:
          - Add a second Optimize instance of the docs plugin
          - Add a second Optimize set of sidebars for the new instance.
          - Configure sidebars for consistent cross-plugin navigation
          - Move one doc (`what-is-optimize`) into new instance
          - Update links in documentation for the moved doc (and docs that reference it)
     2. ✅ Iteratively move remaining "next" Optimize docs over: #1204 and #1214

2. ✅ Create a "3.8.0" Optimize version
    1.  ✅ #1234
        This will be merged into `main` all at once, because this is the current version of documentation and we will confuse people if we release it in pieces.
        - Configure the 3.8.0 version of Optimize documentation
        - Customize NavbarItem so that the version dropdown shows the correct versions, based on which docs are being viewed (in #1235)
        - Extract everything from docs/8.0/components/optimize to optimize/3.8.0 (in #1236)
        - Add rewrite rules for optimize/3.8.0 docs (in #1238)
        - Revert #1193 (because we want these things in the sitemap now) (in #1239)
3. ✅ #1286
    For consistency with the new folders planned to move, we'll want to move all the existing Optimize docs into a `/components/` folder.
    1. Move `/optimize/*` into `/optimize/components/*` for vNext
    2. Move `/optimize_versioned_docs/version-3.8.0/*` into `/optimize_versioned_docs/version-3.8.0/*` for v3.8.0
4. ✅  Move Optimize docs from apis-clients & self-managed sections into Optimize instance
    1. ✅  vNext: #1293
       1. Move `/docs/apis-clients/optimize-api/` docs into vNext 
       2. Move `/docs/self-managed/optimize-deployment/` docs into vNext
   2. ✅  v3.8.0: #1300 
      1. Move `/docs/apis-clients/optimize-api/` docs into 3.8.0 
      2. Move `/docs/self-managed/optimize-deployment/` docs into 3.8.0 
5. ✅ #1217 
6. ✅ Create a "3.7.0" Optimize version
    - ✅ #1408 

### Follow-ups/polish

- ✅ #1313
- #1333 
- Make the "Components" item in the top nav active when in the Optimize docs
- Make footer links include cross-instance docs ($docs$ skips over $optimize$ pages, and vice versa)
- #1396

## POC

See #906 for a POC of this work.

## Decision to make

What do we want to use as the root directory for the new section of documentation? 

In #906, I used `optimize` as the root. It’s arbitrary, and we can choose whatever we want (except for something starting with `docs` since that is the main docs instance). (**Update:** we chose `optimize`.)

## Background
This is Recommendation #1 in [the Optimize Multi-Instance Documentation Proposal](https://docs.google.com/document/d/1GpB7OcSzo5htcBHDAdEVAOgqMOAWrAaJELycz5Gqcko/edit#).

**Comment by akeller** [comment]
@pepopowitz wants a post mortem for this, and I 100% agree.

**Comment by pepopowitz** [comment]
> @pepopowitz wants a post mortem for this, and I 100% agree.

Added to our 1:1 board 😅

**Comment by akeller** [comment]
@pepopowitz what work remains open that must be completed for this issue? I know other items have popped up but they seem outside of the scope of this epic. 

(More context: I'm interested in pivoting our focus to the C7 SEO issues.)

**Comment by pepopowitz** [comment]
this is the last bit I consider part of the original work!

<img width="705" alt="image" src="https://user-images.githubusercontent.com/1627089/196992008-e2a024c5-175c-4734-91ac-5c4f15c81a71.png">

**Comment by pepopowitz** [comment]
Before I close it -- how much do we care about these 2 issues? Are they worth creating issues for? 

> * Make the "Components" item in the top nav active when in the Optimize docs
> * Make footer links include cross-instance docs ($docs$ skips over $optimize$ pages, and vice versa) 
>

**Comment by pepopowitz** [comment]
👍 

CLOSING AS COMPLETED

<img src="https://media2.giphy.com/media/o75ajIFH0QnQC3nCeD/giphy.gif"/>
